tests/mpi/communicator_duplicated.cc

```diff
@@ -0,0 +1,91 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// check MPI::duplicate_communicator, free_communicator, and
+// DuplicatedCommunicator
+
+
+#include <deal.II/base/mpi.h>
+
+#include "../tests.h"
+
+
+void test(const MPI_Comm comm)
+{
+  int tag = 12345;
+
+  const auto my_rank = Utilities::MPI::this_mpi_process(comm);
+
+  MPI_Comm comm2 = Utilities::MPI::duplicate_communicator(comm);
+  Utilities::MPI::DuplicatedCommunicator pcomm3(comm);
+
+  if (my_rank==1)
+    {
+      int value[3]={1,2,3};
+      int dest = 0;
+      
+      MPI_Send(&value[0],
```

Comment:
I would use `MPI_Isend` here, since this is what you want (and what you get since you send only one double)...

Reply:
you are right, I am surprised this test works correctly this way.

Reply:
It works because you are only sending a small amount of data (here 1 int). For small payload sizes the MPI implementations will copy the data into a internal buffer and then send. The consequence is that the user has the impression of `Isend`. However, if you would increase the size of payload you will notice that it will deadlock once you have reached a threshold, from which the MPI implementation do not buffer the data any more...